### DISCUSSION TITLE
custom package folders not loading.

### DISCUSSION THREAD

**Discussion by Vercjames** [vercel/turborepo#9352] [Help] [answered]
### Summary

I'm trying to understand how the `packages` directory works when using the repo. I can't see to get custom folders to work. I've create an example project.

https://github.com/Vercjames/Turbo-Issues

Where I've inverted the `eslint-config` to `config-eslint` to test toggling between different coding styles. feel free to toggle between the 2 lint files. 

```
/** @type {import("eslint").Linter.Config} */
module.exports = {
  root: true,
  extends: [
    // QUESTION: Why does this line fail
    "@repo/config-eslint/next.js",
    // QUESTION: While this line works
    // "@repo/eslint-config/next.js"
  ],
  parser: "@typescript-eslint/parser",
  parserOptions: {
    project: true,
  },
};
```

I'm hoping I just missed a simple config, but I made sure to update the `apps/web` package.json to include the new folder. heres the error I'm getting:

Problem with ESLint reformatting: `page.tsx`: Error: Failed to load config `"@repo/config-eslint/next.js"` to extend from.

where my page.tsx is the file I'm attempting to reformat. 

### Additional information

```js
Mac
Node:v20.12.2
npm: 10.8.2
yarn: 1.22.22
```

### Example

https://github.com/Vercjames/Turbo-Issues

**Comment by anthonyshew** [comment]
In the `@repo/config-eslint/next.js`, `@repo/config-eslint` refers to the name of the package. You would also need to change the name of the package in the `name` field of the package's `package.json` to correspond to the new name. If you do change names, make sure you run your package manager's install command, too, so that your lockfile is updated to reflect the change.

**Reply by Vercjames** [reply]
The example app already matched your described fixed. If I'm missing something, is there any chance you can take a look at the example app directly? the condition of "You would also need to change the name of the package in the name field of the package's package.json" was already met prior to making this post. 

@repo/config-eslint
```json
{
  "name": "@repo/config-eslint",
  "version": "0.0.0",
  "private": true,
  "files": [
    "library.js",
    "next.js",
    "react-internal.js"
  ],
  "devDependencies": {
    "@vercel/style-guide": "^5.2.0",
    "eslint-config-turbo": "^2.0.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-only-warn": "^1.1.0",
    "@typescript-eslint/parser": "^7.1.0",
    "@typescript-eslint/eslint-plugin": "^7.1.0",
    "typescript": "5.5.4"
  }
}
```
vs

@repo/eslint-config
```json
{
  "name": "@repo/eslint-config",
  "version": "0.0.0",
  "private": true,
  "files": [
    "library.js",
    "next.js",
    "react-internal.js"
  ],
  "devDependencies": {
    "@vercel/style-guide": "^5.2.0",
    "eslint-config-turbo": "^2.0.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-only-warn": "^1.1.0",
    "@typescript-eslint/parser": "^7.1.0",
    "@typescript-eslint/eslint-plugin": "^7.1.0",
    "typescript": "5.5.4"
  }
}
```

**Reply by anthonyshew** [reply, marked as answer]
Sorry, I forgot about this specific constraint of ESLint: https://github.com/eslint/eslint/discussions/15828

It forces you to have the `eslint-config` prefix.
